RealmSwift/Query.swift

```diff
@@ -1035,7 +1035,8 @@ private func buildPredicate(_ root: QueryNode, subqueryCount: Int = 0) -> (Strin
         case .mapAnySubscripts(let keyPath, let keys):
             build(keyPath)
             for key in keys {
-                formatStr.append("[%@]")
+                let arg = (key as? String == "#any") ? "[%K]" : "[%@]"
```

Comment:
This is incorrectly turning `.where { $0.anyCol["#any"] == ... }` into `.where { $0.anyCol.any == ... }`. The entire point of all this is that `.any` *doesn't* produce something that could be a string subscript.